crates/rattler_repodata_gateway/src/gateway/subdir.rs

```diff
@@ -107,40 +108,28 @@ impl SubdirData {
             }
         };
 
-        // At this point we have exclusive write access to this specific entry. All other tasks
-        // will find a pending entry and will wait for the records to become available.
+        // At this point we have exclusive write access to this specific entry. All
+        // other tasks will find a pending entry and will wait for the records
+        // to become available.
         //
-        // Let's start by fetching the records. If an error occurs we immediately return the error.
-        // This will drop the sender and all other waiting tasks will receive an error.
-        let records = match tokio::spawn({
```

Comment:
I removed the spawn here because I dont think its needed. All implementations fan out to spawn blocking when blocking operations are performed.